homeassistant/components/motion_blinds/sensor.py

```diff
@@ -183,7 +183,11 @@ def available(self):
         if self.coordinator.data is None:
             return False
 
-        if not self.coordinator.data[KEY_GATEWAY][ATTR_AVAILABLE]:
+        gateway_available = self.coordinator.data[KEY_GATEWAY][ATTR_AVAILABLE]
+        if self._device_type == TYPE_GATEWAY:
+            return gateway_available
+
+        if not gateway_available:
             return False
 
         return self.coordinator.data[self._device.mac][ATTR_AVAILABLE]
```

Comment:
```suggestion
        return (
            gateway_available
            and self.coordinator.data[self._device.mac][ATTR_AVAILABLE]
        )
```

Reply:
Will this also work if gateway_available is False and [self._device.mac][ATTR_AVAILABLE] is not in the dict, or will that give an KeyError?

Reply:
just tested it, you are right